tests/unit/unit-solving-interface.cpp

```diff
@@ -66,16 +66,22 @@ TEST_P(UnitSolveTests, CheckSatAssuming)
     // mathsat is the only solver (so far)
     // to have the restriction that assumptions must be
     // (negated) boolean constants
-    EXPECT_EQ(s->get_solver_enum(), MSAT);
+    // Generic solvers are SMT-LIB compliant, and threfore
```

Comment:
```suggestion
    // Generic solvers are SMT-LIB compliant, and therefore
```

Reply:
quick question -- I thought they wouldn't actually need to follow this restriction because you use define-funs and those are allowed in `check-sat-assuming`. Is that wrong?

Reply:
huh, interesting, let me check.